src/otx/algo/detection/backbones/backbone_factory.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Backbone factory for detection."""
+
+from torch import nn
+
+
+class DetectionBackboneFactory:
```

Comment:
I had a similar impression with Eunwoo, isn't the existing way better than using multiple if statements without unification?
I don't think it's a good thing that we have to modify this backbone factory class every time we add a backbone.

To be honest, I don't know why we need this class.

Reply:
Remove this f38b613